analytics/analytics/core/models/dimensions.py

```diff
@@ -174,6 +206,11 @@ class Meta:
         db_comment="Whether this job has 'No need to rebuild' in its trace.",
     )  # type: ignore
 
+    # This field is duplicated here (from the gitlab job data dimension)
+    # to allow for the generated `infrastructure_error` field
+    job_type = models.CharField(
+        max_length=max(len(c) for c, _ in JobType.choices), choices=JobType.choices
+    )
```

Comment:
We could consider using a database trigger to set `infrastructure_error` instead of duplicating this column. Something to think about for a follow up at least; there's already substantial DB changes happening in this PR so I don't think it's worth adding more complexity unless you feel otherwise

Reply:
We could do that as a follow up, although I'll just point out that duplicate data isn't a problem at all in this type of database. Since these records aren't subject to modification like they would be in a normal web application (or if they are it's a system wide modification applied consistenly), it's not an issue.